lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp

```diff
@@ -1407,19 +1407,29 @@ bool ConsumeIncludeOption(StringRef &arg, StringRef &prefix) {
                                 "-isystem",
                                 "-isysroot",
                                 "-ivfsoverlay"};
+
   for (StringRef &option : options)
     if (arg.consume_front(option)) {
       prefix = option;
       return true;
     }
+
+  // We need special handling for -fmodule-file as we need to
+  // split on the final = after the module name.
+  if (arg.startswith("-fmodule-file=")) {
+    prefix = arg.substr(0, arg.rfind("=") + 1);
```

Comment:
I assume this is safe, even if arg is either "foo",  or "foo="?

Reply:
The two valid cases are `-fmodule-file=some/path` and `-fmodule-file=modulename=some/path`. We know the first `=` is present as we checked the prefix above, so this will split correctly in both cases. There is no form `-fmodule-file some/path` so we don't need to handle that case.